boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -628,4 +628,120 @@ TEST_F(BmsStateMachineTest, check_state_transition_from_fault_to_init_with_air_n
     ASSERT_EQ(CANMSGS_BMS_STATE_MACHINE_STATE_INIT_CHOICE, App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
 
-} // namespace StateMachineTest
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_undervoltage)
+{
+    SetInitialState(App_GetDriveState());
+
+    get_min_cell_voltage_fake.return_val = 2.9f;
+
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_overvoltage)
+{
+    SetInitialState(App_GetInitState());
+
+    get_max_cell_voltage_fake.return_val = 4.3f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_overtemp_drive_state)
+{
+    SetInitialState(App_GetInitState());
+
+    // In Discharge state, acceptible temp range is (-20, 60), should be unaffected by temp of 46 C
+    get_max_temp_degc_fake.return_val = 46.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_NE(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+
+    LetTimePass(state_machine, 1000);
+
+    get_max_temp_degc_fake.return_val = 61.0f;
+    LetTimePass(state_machine, 10);
+
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_overtemp_charge_state)
+{
+    SetInitialState(App_GetChargeState());
+
+    // In Charge state acceptible temp range is (0, 45)
+    get_max_temp_degc_fake.return_val = 46.0f;
```

Comment:
this passed without setting the Is_Charger_connected to true?

Reply:
is_charger_connected.fake.return_val is set to true in the setup function, but it turned out the state machine was changing to fault state for reasons other than those checked by the test. I've added some test-specific setup conditions such that fault state transitions should be triggered by the condition under test.